AdblockPlusSafari/AdblockPlusExtras.m

```diff
@@ -329,4 +287,28 @@ - (BOOL)fileManager:(NSFileManager *)fileManager shouldProceedAfterError:(NSErro
 
 #pragma mark - Private
 
+- (void)onApplicationWillEnterForegroundNotification:(NSNotification *)notification
+{
+  if (self.reloading) {
+    return;
+  }
+
+  __weak __typeof(self) wSelf = self;
+  NSDate *lastActivity = wSelf.lastActivity;
+  wSelf.performingActivityTest = YES;
+  [SFContentBlockerManager reloadContentBlockerWithIdentifier:self.contentBlockerIdentifier completionHandler:^(NSError *error) {
+    dispatch_async(dispatch_get_main_queue(), ^{
+      NSLog(@"%@", error);
+      wSelf.performingActivityTest = YES;
+      [wSelf checkActivatedFlag:lastActivity];
+    });
+  }];
+}
+
+- (void)checkActivatedFlag:(NSDate *)lastActivity
+{
+  [self synchronize];
+  self.activated = !!self.lastActivity && (!lastActivity || [self.lastActivity compare:lastActivity] == NSOrderedDescending);
```

Comment:
Here I check if content blocker is enabled. `Synchronize` is quite important here, since it forces host app reload shared file